homeassistant/components/minecraft_server/sensor.py

```diff
@@ -0,0 +1,265 @@
+            icon=ICON_PROTOCOL_VERSION,
+            unit=UNIT_PROTOCOL_VERSION,
+        )
+
+    async def async_update(self) -> None:
+        """Update protocol version."""
+        self._state = self._server.protocol_version
+
+
+class MinecraftServerLatencyTimeSensor(MinecraftServerSensorEntity):
+    """Representation of a Minecraft Server latency time sensor."""
+
+    def __init__(self, hass: HomeAssistantType, server: MinecraftServer) -> None:
+        """Initialize latency time sensor."""
+        super().__init__(
+            hass=hass,
+            server=server,
+            type_name=NAME_LATENCY_TIME,
+            icon=ICON_LATENCY_TIME,
+            unit=UNIT_LATENCY_TIME,
+        )
+
+    async def async_update(self) -> None:
+        """Update latency time."""
+        self._state = self._server.latency_time
+
+
+class MinecraftServerPlayersOnlineSensor(MinecraftServerSensorEntity):
+    """Representation of a Minecraft Server online players sensor."""
+
+    def __init__(self, hass: HomeAssistantType, server: MinecraftServer) -> None:
+        """Initialize online players sensor."""
+        super().__init__(
+            hass=hass,
+            server=server,
+            type_name=NAME_PLAYERS_ONLINE,
+            icon=ICON_PLAYERS_ONLINE,
+            unit=UNIT_PLAYERS_ONLINE,
+        )
+
+    async def async_update(self) -> None:
+        """Update online players."""
+        self._state = self._server.players_online
+
+
+class MinecraftServerPlayersMaxSensor(MinecraftServerSensorEntity):
+    """Representation of a Minecraft Server maximum number of players sensor."""
+
+    def __init__(self, hass: HomeAssistantType, server: MinecraftServer) -> None:
+        """Initialize maximum number of players sensor."""
+        super().__init__(
+            hass=hass,
+            server=server,
+            type_name=NAME_PLAYERS_MAX,
+            icon=ICON_PLAYERS_MAX,
+            unit=UNIT_PLAYERS_MAX,
+        )
+
+    async def async_update(self) -> None:
+        """Update maximum number of players."""
+        self._state = self._server.players_max
+
+
+class MinecraftServerPlayersListSensor(MinecraftServerSensorEntity):
+    """Representation of a Minecraft Server players list sensor."""
+
+    def __init__(self, hass: HomeAssistantType, server: MinecraftServer) -> None:
+        """Initialize players list sensor."""
+        super().__init__(
+            hass=hass,
+            server=server,
+            type_name=NAME_PLAYERS_LIST,
+            icon=ICON_PLAYERS_LIST,
+            unit=UNIT_PLAYERS_LIST,
+        )
+
+    async def async_update(self) -> None:
+        """Update players list."""
+        players_list = self._server.players_list
+        players_string = None
+
+        if players_list is not None:
+            if not players_list:
+                players_string = "[]"
+            else:
+                separator = ", "
+                players_string = f"[{separator.join(players_list)}]"
```

Comment:
How will this sensor be used? Maybe instead use the number of players as the sensor state and return the list of players unmodified as a state attribute that can be extracted if wanted by a template sensor.

Reply:
Like the description sensor I was thinking that this can be used easier for example in a notification. But having it instead as a state attribute in the number of players sensor is for sure also possible. Should I go that way?

Reply:
Yes, please. :+1: